src/sidebar/components/annotation-action-bar.js

```diff
@@ -50,9 +50,9 @@ function AnnotationActionBar({
   const showDeleteAction = userIsAuthorizedTo('delete');
   const showEditAction = userIsAuthorizedTo('update');
 
-  // Anyone may flag an annotation except the annotation's author.
-  // This option is even presented to anonymous users
-  const showFlagAction = userProfile.userid !== annotation.user;
+  //  Only authenticated users can flag an annotation, except the annotation's author.
+  const showFlagAction =
+    !!userProfile.userid && userProfile.userid !== annotation.user;
```

Comment:
Since `userid` is either `null` or a non-empty string, the boolean cast isn't strictly needed here.

Reply:
You are correct. I intended to make the type of  `showFlagAction` always `boolean`, no `boolean | null`, although in this particular case doesn't make a difference.